src/Symfony/Bundle/FrameworkBundle/Command/AbstractConfigCommand.php

```diff
@@ -83,4 +89,14 @@ public function validateConfiguration(ExtensionInterface $extension, $configurat
             throw new \LogicException(sprintf('Configuration class "%s" should implement ConfigurationInterface in order to be dumpable', get_class($configuration)));
         }
     }
+
+    private function initializeExtension($bundle, $extension)
+    {
+        $parameters = $this->getContainer()->getParameterBag()->all();
+        $container = new ContainerBuilder(new ParameterBag($parameters));
+        $container->registerExtension($extension);
+
+        $bundle = clone $bundle;
+        $bundle->build($container);
```

Comment:
This won't solve the issue entirely actually. The whole point of extensions having a config which depends on the bundle build method is to allow *other* bundles than themselves to apply changes on it (for instance, DoctrineBundle registers the support of the entity type)